File: backend/src/main/java/com/br/ccps/model/Veterinario.java
package com.br.ccps.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Veterinario {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    private String nome;

    @Column(length = 20, unique = true)
    private String crmv;
    
    private String cpf;
    
    private LocalDate dataNascimento;

    private String email;

    private String telefone;

    @OneToMany(mappedBy = "veterinario")
    private List<Ccps> ccpsList;
}

File: frontend/src/models/Veterinario.ts
import { Ccps } from "./Ccps";

export type Veterinario = {
    id: string;
    nome: string;
    crmv: string;
    cpf: string;
    dataNascimento: string; 
    email: string;
    senha: string;
    telefone: string;
    ccpsList: Ccps[];
  }